packages/dappmanager/src/calls/portsScanGet.ts

```diff
@@ -0,0 +1,17 @@
+import { PortScanResponse } from "../types";
+import { getPortsScan } from "../utils/getPortsScan";
+import * as db from "../db";
+
+export async function portsScanGet(): Promise<PortScanResponse[]> {
+  const packagePort = db.portsToOpen.get();
+  // udp ports cannot be scanned
+  // as a better approach, if tcp ports were opened with UpNp, udp quite probably were opened as well
+  const tcpPorts = packagePort.filter(port => port.protocol === "TCP");
```

Comment:
This formating should be done in the utility `getPortsScan` to keep things together. This way you can use a fully typed interface instead of `ports: string[]` which actually means only the TCP port number in string form